src/main/java/com/suse/saltstack/netapi/client/SaltStackHttpClientConnectionFactory.java

```diff
@@ -0,0 +1,16 @@
+package com.suse.saltstack.netapi.client;
+
+import com.suse.saltstack.netapi.config.SaltStackClientConfig;
+
+/**
+ * Implementation of a factory for connections using Apache's HttpClient.
+ *
+ * @see com.suse.saltstack.netapi.client.SaltStackHttpClientConnection
+ */
+public class SaltStackHttpClientConnectionFactory implements SaltStackConnectionFactory {
+    @Override
+    public SaltStackHttpClientConnection create(String endpoint,
+                                         SaltStackClientConfig config) {
```

Comment:
Please do not use such block indentation, rather indent 8 spaces after breaking a long line (as [mentioned] (https://github.com/SUSE/saltstack-netapi-client-java/pull/1#issuecomment-72463963)).